File: Python3/HashTable.py
import sys
import math

class HashTable:
    def __init__(self, size) :
        self.size = size ;
        self.vals = [None for i in range(self.size)];

    def hash(self,key) :
        A = (math.sqrt(5) - 1 ) / 2 

        return math.floor(self.size * ((key * A) % 1));
    
    def add(self, key , val):
        hashedKey = self.hash(key)
        self.vals[hashedKey] = val;
        return True
    
    def delete(self, key) :
        hashedKey = self.hash(key)
        self.vals[hashedKey] = None;
    
    def get(self,key) :
        hashedKey = self.hash(key)
        if self.vals[hashedKey] == None:
            return False;
        else :
            return self.vals[hashedKey];
    

table = HashTable(10);

table.add(123, "apple")
table.add(432, "mango")
table.add(213, "banana")
table.add(654, "guava")

print(table.vals)

table.delete(123)

print(table.vals)

File: Swift/Data Structures .playground/Sources/HashTable.swift
import Foundation


public class HashTable {
    
    var size : Int;
    var vals : [Any?]!
    
    public init(size : Int) {
        self.size = size ;
        self.vals = [];
        for i in 0...self.size{
            self.vals.append(nil)
        }
    }
    
    private func hash(key : Int) -> Double {
        let A = (sqrt(5) - 1) / 2 ;
        let hashedKey = Double(self.size) * ((Double(key) * A).truncatingRemainder(dividingBy: Double(1)))
        return floor(hashedKey)
    }
    
    public func add(key : Int , val : Any)  {
        let hashedKey = Int(self.hash(key: key))
        self.vals[hashedKey] = val;
        
    }
    
    
    public func delete(key : Int) {
        let hashedKey = Int(self.hash(key: key))
        self.vals[hashedKey] = nil
        
    }
    
    public func get(key: Int) -> Any {
        let hashedKey = Int(self.hash(key: key))
        if self.vals[hashedKey] == nil {
            return -1
        }
        return self.vals[hashedKey]!
    }
    
    public func printt() {
        print(self.vals)
    }
    
}
